### PR TITLE
Add deploy.js improvements and fix some bugs

### PR DESCRIPTION
**Pull request by cal-a-mari**

### CODE REVIEW

File: src/components/CreateListingModal.tsx [hunk 1]
```diff
@@ -31,14 +32,13 @@ function CreateListingModal(props: Props) {
...
         const contract = KasuContract();
-
         try {
             contract.listNFT(
                 props.tokenID,
                 props.tokenAddress,
                 formValues.rentalDuration,
                 formValues.interestRate,
-                formValues.collateralRequired
+                ethers.utils.parseEther(formValues.collateralRequired)
```
Comment (cal-a-mari) [inline src/components/CreateListingModal.tsx:43]: @tedwu13 fyi the rest of the app/smart contract assumes this is in Wei so to make it easier I'm gonna just pass in the wei value

Comment (tedwu13) [reply src/components/CreateListingModal.tsx:43]: yeah makes sense thanks for converting it

File: scripts/deploy.js [hunk 2]
```diff
@@ -30,24 +30,53 @@ async function main() {
...
+    const account1 = accounts[1];
+    console.log(`Borrowing some NFTs from account ${account1.address}`)
+    const listings = await kasuContract.viewAllListings();
+    for (let i = 0; i < listings.length - 2; i++) {
+        const listing = listings[i];
+        const collateralRequiredInEth = Number(hre.ethers.utils.formatEther(listing.collateralRequired));
+        console.log(`Listing id: ${listing.id}, rent duration: ${listing.duration}, collateral req: ${collateralRequiredInEth} eth, interest rate: ${listing.dailyInterestRate}%`)
+        const totalAmountRequiredInEth =
+            collateralRequiredInEth  +
+            (collateralRequiredInEth * listing.duration * (listing.dailyInterestRate / 100));
```
Comment (forcey) [inline scripts/deploy.js:67]: I'm getting an error
```
 Contract call:       Kasu#borrow
  From:                0x70997970c51812dc3a010c7d01b50e0d17dc79c8
  To:                  0x5fbdb2315678afecb367f032d93f642f64180aa3
  Value:               7.6999 ETH

  Error: VM Exception while processing transaction: reverted with reason string 'must send the exact payment'
```
I think you need to do the whole thing in BigNumber... see https://github.com/forcey/nft-rental-marketplace/blob/main/src/components/BorrowModal.tsx#L14, it's actually not very convoluted.

Comment (cal-a-mari) [reply scripts/deploy.js:67]: @forcey ugh my bad. I tried to use this expression using bignumbers but got number underflow due to multiplying a value that's < 1. Didn't realize the expression you have also would work. Just pushed a fix for this.

**Pull request merged**